File: backend/src/main/java/at/ftmahringer/wahlsystem/entity/Candidate.java
package at.ftmahringer.wahlsystem.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "candidates")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Candidate {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "first_name", nullable = false)
    private String firstName;

    @Column(name = "last_name", nullable = false)
    private String lastName;

    @Column(name = "class_name")
    private String className;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "sort_order")
    @Builder.Default
    private Integer sortOrder = 0;

    @Column(name = "active")
    @Builder.Default
    private Boolean active = true;

    @Column(name = "vote_count")
    @Builder.Default
    private Integer voteCount = 0;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "election_id", nullable = false)
    private Election election;
}

File: frontend/src/types/candidate.ts
export interface Candidate {
  id: number;
  firstName: string;
  lastName: string;
  className: string | null;
  description: string | null;
  sortOrder: number;
  active: boolean;
  electionId: number;
}

export interface CreateCandidateRequest {
  firstName: string;
  lastName: string;
  className?: string;
  description?: string;
  sortOrder?: number;
  electionId: number;
}
